pilot/pkg/networking/core/httproute.go

```diff
@@ -627,6 +627,14 @@ func generateVirtualHostDomains(service *model.Service, listenerPort int, port i
 		all = append(all, a.Hostname.String())
 	}
 	for _, s := range all {
+		// Check if 's' is an IP address. We don't add trailing dots to IPs.
+		isIP := net.ParseIP(s) != nil
+		// Add the absolute FQDN variant (with trailing dot) if it's not an IP address.
```

Comment:
nit: does this make sense under GenerateAltVirtualHosts since its basically an alt host?

Reply:
I think it makes sense to move there